enclave/src/rpc/polkadex_rpc_gateway.rs

```diff
@@ -16,20 +16,56 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-use crate::polkadex_balance_storage::{lock_storage_and_get_balances, Balances};
-use crate::polkadex_gateway::{authenticate_user, GatewayError};
-use polkadex_sgx_primitives::{AccountId, AssetId};
+pub extern crate alloc;
+use alloc::{string::String, string::ToString};
+
+use crate::polkadex_balance_storage::{
+    lock_storage_and_get_balances, lock_storage_and_withdraw, Balances,
+};
+use crate::polkadex_gateway::{authenticate_user, cancel_order, place_order, GatewayError};
+use crate::rpc::rpc_info::RpcCallStatus;
+use polkadex_sgx_primitives::types::{Order, OrderUUID};
+use polkadex_sgx_primitives::{AccountId, AssetId, Balance};
 use sgx_types::SgxResult;
+use substratee_stf::{TrustedCall, TrustedOperation};
 
 /// Gateway trait from RPC API -> Polkadex gateway implementation
 pub trait RpcGateway: Send + Sync {
+    /// verifies that the proxy account (if any) is authorized to represent the main account
     fn authorize_user(
         &self,
         main_account: AccountId,
         proxy_account: Option<AccountId>,
     ) -> Result<(), GatewayError>;
 
+    /// verifies that the proxy account (if any) is authorized to represent the main account
+    /// given a trusted call inside a trusted operation (convenience function)
+    fn authorize_trusted_call(
+        &self,
+        trusted_operation: TrustedOperation,
+    ) -> Result<TrustedCall, String>;
+
+    /// get the balance of a certain asset ID for a given account
     fn get_balances(&self, main_account: AccountId, asset_it: AssetId) -> SgxResult<Balances>;
+
+    /// place an order
+    fn place_order(
+        &self,
+        main_account: AccountId,
+        proxy_acc: Option<AccountId>,
+        order: Order,
+    ) -> Result<OrderUUID, GatewayError>;
```

Comment:
This will be Result<(),Error>; as we will be working with non blocking calls. But this can be changed later on.